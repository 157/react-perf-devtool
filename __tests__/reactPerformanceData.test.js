var { getReactPerformanceData } = require('../src/shared/parse')
var MEASURES = require('../samples/measures')

describe('React Performance Data', () => {
  test('Parse component name and phase name', () => {
    console.log(getReactPerformanceData(MEASURES))
    expect(getReactPerformanceData(MEASURES)).toMatchSnapshot()
  })
})
