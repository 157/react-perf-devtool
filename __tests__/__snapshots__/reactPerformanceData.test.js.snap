// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Performance Data Parse component name and phase name 1`] = `
Object {
  "App": Object {
    "componentDidMount": Object {
      "timeSpent": Array [
        1.2199999999999136,
      ],
    },
    "componentDidUpdate": Object {
      "timeSpent": Array [],
    },
    "componentWillMount": Object {
      "timeSpent": Array [
        3.1499999999999773,
      ],
    },
    "componentWillReceiveProps": Object {
      "timeSpent": Array [],
    },
    "componentWillUnmount": Object {
      "timeSpent": Array [],
    },
    "componentWillUpdate": Object {
      "timeSpent": Array [],
    },
    "mount": Object {
      "timeSpent": Array [
        58.59999999999991,
      ],
    },
    "render": Object {
      "timeSpent": Array [],
    },
    "shouldComponentUpdate": Object {
      "timeSpent": Array [],
    },
    "unmount": Object {
      "timeSpent": Array [],
    },
    "update": Object {
      "timeSpent": Array [],
    },
  },
  "Introduction": Object {
    "componentDidMount": Object {
      "timeSpent": Array [
        0.5099999999999909,
      ],
    },
    "componentDidUpdate": Object {
      "timeSpent": Array [
        0.849999999999909,
      ],
    },
    "componentWillMount": Object {
      "timeSpent": Array [
        0.8000000000000114,
      ],
    },
    "componentWillReceiveProps": Object {
      "timeSpent": Array [],
    },
    "componentWillUnmount": Object {
      "timeSpent": Array [],
    },
    "componentWillUpdate": Object {
      "timeSpent": Array [],
    },
    "mount": Object {
      "timeSpent": Array [
        14.260000000000048,
      ],
    },
    "render": Object {
      "timeSpent": Array [],
    },
    "shouldComponentUpdate": Object {
      "timeSpent": Array [],
    },
    "unmount": Object {
      "timeSpent": Array [],
    },
    "update": Object {
      "timeSpent": Array [
        1.9149999999999636,
      ],
    },
  },
  "Main": Object {
    "componentDidMount": Object {
      "timeSpent": Array [
        1.6650000000000773,
      ],
    },
    "componentDidUpdate": Object {
      "timeSpent": Array [],
    },
    "componentWillMount": Object {
      "timeSpent": Array [
        2.1999999999999886,
      ],
    },
    "componentWillReceiveProps": Object {
      "timeSpent": Array [],
    },
    "componentWillUnmount": Object {
      "timeSpent": Array [],
    },
    "componentWillUpdate": Object {
      "timeSpent": Array [],
    },
    "mount": Object {
      "timeSpent": Array [
        53.05000000000001,
      ],
    },
    "render": Object {
      "timeSpent": Array [],
    },
    "shouldComponentUpdate": Object {
      "timeSpent": Array [],
    },
    "unmount": Object {
      "timeSpent": Array [],
    },
    "update": Object {
      "timeSpent": Array [],
    },
  },
  "Message": Object {
    "componentDidMount": Object {
      "timeSpent": Array [
        0.8949999999999818,
      ],
    },
    "componentDidUpdate": Object {
      "timeSpent": Array [],
    },
    "componentWillMount": Object {
      "timeSpent": Array [
        1.2250000000000227,
      ],
    },
    "componentWillReceiveProps": Object {
      "timeSpent": Array [],
    },
    "componentWillUnmount": Object {
      "timeSpent": Array [],
    },
    "componentWillUpdate": Object {
      "timeSpent": Array [],
    },
    "mount": Object {
      "timeSpent": Array [
        8.740000000000066,
      ],
    },
    "render": Object {
      "timeSpent": Array [],
    },
    "shouldComponentUpdate": Object {
      "timeSpent": Array [],
    },
    "unmount": Object {
      "timeSpent": Array [],
    },
    "update": Object {
      "timeSpent": Array [],
    },
  },
  "Posts": Object {
    "componentDidMount": Object {
      "timeSpent": Array [
        2.07000000000005,
      ],
    },
    "componentDidUpdate": Object {
      "timeSpent": Array [],
    },
    "componentWillMount": Object {
      "timeSpent": Array [
        1.6000000000000227,
      ],
    },
    "componentWillReceiveProps": Object {
      "timeSpent": Array [],
    },
    "componentWillUnmount": Object {
      "timeSpent": Array [],
    },
    "componentWillUpdate": Object {
      "timeSpent": Array [],
    },
    "mount": Object {
      "timeSpent": Array [
        6.535000000000025,
      ],
    },
    "render": Object {
      "timeSpent": Array [],
    },
    "shouldComponentUpdate": Object {
      "timeSpent": Array [],
    },
    "unmount": Object {
      "timeSpent": Array [],
    },
    "update": Object {
      "timeSpent": Array [],
    },
  },
  "Projects": Object {
    "componentDidMount": Object {
      "timeSpent": Array [
        0.20000000000004547,
      ],
    },
    "componentDidUpdate": Object {
      "timeSpent": Array [],
    },
    "componentWillMount": Object {
      "timeSpent": Array [
        4.044999999999959,
      ],
    },
    "componentWillReceiveProps": Object {
      "timeSpent": Array [],
    },
    "componentWillUnmount": Object {
      "timeSpent": Array [],
    },
    "componentWillUpdate": Object {
      "timeSpent": Array [],
    },
    "mount": Object {
      "timeSpent": Array [
        17.335000000000036,
      ],
    },
    "render": Object {
      "timeSpent": Array [],
    },
    "shouldComponentUpdate": Object {
      "timeSpent": Array [],
    },
    "unmount": Object {
      "timeSpent": Array [],
    },
    "update": Object {
      "timeSpent": Array [
        7.100000000000023,
      ],
    },
  },
}
`;
